source/Player.hx

```diff
@@ -187,6 +210,48 @@ import flixel.group.FlxGroup;
     btmBox.y = y + height - hitBoxHeight;
   }
 
+  //Override player.kill to drop any weilded items
+  override public function kill():Void 
+  {
+	  if (weilding){
+		dropCurrentEquip();
+	  }
+	  super.kill();
+	  new FlxTimer().start(2, resetLevel, 1); //This helps speed things up for debugging
+  }
+  
+  private function resetLevel(Timer:FlxTimer):Void
```

Comment:
Moved the function itself to the playstate, but I still kept the override in the player class. I access the playstate method through FlxG the way you suggested I access the player class earlier.